### PR TITLE
chore(docs): move ILP details from insert data to ilp overview

### PR DESCRIPTION
**Pull request by amyshwang**
Preview - https://deploy-preview-1411--unruffled-blackwell-31bfb2.netlify.app/

### CODE REVIEW

**Comment by amyshwang** [comment]
Cf error handling is repeated here - https://questdb.io/docs/reference/api/ilp/tcp-receiver/#error-handling

File: docs/develop/insert-data.md [hunk 1]
```diff
@@ -41,12 +41,6 @@ In summary, these are the different options:
   - SQL `INSERT` statements.
   - Use `curl` on the command line.
 
-## InfluxDB Line Protocol
-
-The InfluxDB Line Protocol (ILP) is a text protocol over TCP on port 9009.
-
-It is a one-way protocol to insert data, focusing on simplicity and performance.
-
 Here is a summary table showing how it compares with other ways to insert data
```
Comment (javier) [inline docs/develop/insert-data.md:44]: I believe we need to rephrase this now that is been taken out of the ILP section. Maybe here is a summary table comparing the different ways to insert data we support

File: docs/develop/insert-data.md [hunk 2]
```diff
@@ -57,6 +51,12 @@ that we support:
 | SQL `INSERT` statements  | Transaction-level                | Good                                |
 | SQL `COPY` statements    | Transaction-level                | Suitable for one-off data migration |
 
+## InfluxDB Line Protocol
+
+The InfluxDB Line Protocol (ILP) is a text protocol over TCP on port `9009`.
+
+It is a one-way protocol to insert data, focusing on simplicity and performance.
+
```
Comment (javier) [inline docs/develop/insert-data.md:59]: this comment is for the paragrap saying 

`With sufficient client-side validation, the lack of errors to the client and confirmation isn't necessarily a concern: QuestDB will log out any issues and disconnect on error. The database will process any valid lines up to that point and insert rows.`

I would probably remove it altogether from here and would add it to the "error handling section" of the TCP receiver

**Review by javier — COMMENTED** [review]
I added a couple of comments, but after reading the. new layout, I believe we should probably consolidate the ILP overview and the ILP TCP Receiver under a single page. Otherwise we are going to have a lot of duplication. We already have some, and for example AUTH is not mentioned at the ILP overview, which would be important.

Also, if we refactor the TCP Receiver page, I would move the whole commit section to the bottom as it is deprecated but right now it is in the middle of the page, with current content both above and below, which makes navigation confusing

**Comment by amyshwang** [comment]
The current Overview page:
Screenshot 2023-03-08 at 14 07 22
I think it is a bit too long. I wonder if we should do the following:

Take 'Usage' to the original TCP received page, with:
- Capacity planning
- Commit strategy (some info deprecated since dynamic commit)

This will have an impact on links so I'd like to decide on a strategy before implementing it.

File: docs/reference/api/ilp/overview.md [hunk 3]
```diff
@@ -20,14 +20,50 @@ languages: [ILP client libraries](/docs/reference/clients/overview).
...
-We provide examples in a number of programming languages. See our
-[ILP section](/docs/develop/insert-data#influxdb-line-protocol) of the "develop"
-docs.
+## TCP receiver overview
+
+The TCP receiver is a high-throughput ingestion-only API for QuestDB. Here are
+some key facts about the service:
+
+- ingestion only, there is no query capability
+- accepts plain text input in a form on InfluxDB Line Protocol
```
Comment (javier) [inline docs/reference/api/ilp/overview.md:29]: would that be "a form on" or "a form of"?

File: docs/reference/api/ilp/overview.md [hunk 4]
```diff
@@ -253,3 +292,145 @@ supported value types:
...
+sending data you should be aware of the performed conversions.
+
+See:
+
+- [QuestDB Types in SQL](/docs/reference/sql/datatypes)
+- [ILP types and cast conversion tables](/docs/reference/api/ilp/columnset-types)
+
+### Constructing well-formed messages
+
+Different library implementations will perform different degrees content
```
Comment (javier) [inline docs/reference/api/ilp/overview.md:351]: not sure if we need "of" between "degrees" and "content"

File: docs/reference/api/ilp/overview.md [hunk 5]
```diff
@@ -253,3 +292,145 @@ supported value types:
...
+  additional ones as column values.
+
+- **Don't change column type between rows.**
+
+- **Supply timestamps in order.** These need to be at least equal to previous
+  ones in the same table, unless using the out of order feature. This is not
+  necessary if you use the [out-of-order](/docs/guides/out-of-order-commit-lag)
+  feature.
+
+### Errors handling
```
Comment (javier) [inline docs/reference/api/ilp/overview.md:377]: Should this be singular Error?

File: docs/reference/api/ilp/overview.md [hunk 6]
```diff
@@ -253,3 +292,145 @@ supported value types:
...
+  the timestamp
+- the timestamp is specified as a column instead of appending it to the end of
+  the line
+- timestamp appears as a column and is also present at the end of the line, the
+  value sent as a field will be used
+
+With sufficient client-side validation, the lack of errors to the client and
+confirmation isn't necessarily a concern: QuestDB will log out any issues and
+disconnect on error. The database will process any valid lines up to that point
+and insert rows.
```
Comment (javier) [inline docs/reference/api/ilp/overview.md:419]: Should we link here to how to resume ingestion in the case of errors with WAL backed tables? https://questdb.io/docs/reference/sql/alter-table-resume-wal/

File: docs/reference/api/ilp/tcp-receiver.md [hunk 7]
```diff
@@ -4,75 +4,44 @@ sidebar_label: TCP Receiver
...
 
-It is recommended that sending applications reuse TCP connections. If QuestDB
-receives an invalid message, it will discard invalid lines, produce an error
-message in the logs and forcibly _disconnect_ the sender to prevent further data
-loss.
+- I/O worker pool - `line.tcp.io.worker.count`, threads responsible for handling
+  incoming TCP connections and parsing received ILP messages
+- writer pool - `line.tcp.writer.worker.count`, threads responsible for table
+  writes
+- shared pool - `shared.worker.count`, threads responsible for handling O3 data
```
Comment (javier) [inline docs/reference/api/ilp/tcp-receiver.md:15]: not sure if O3 is so easy to understand, maybe we should say "out-of-order data"?

File: docs/reference/api/ilp/tcp-receiver.md [hunk 8]
```diff
@@ -4,75 +4,44 @@ sidebar_label: TCP Receiver
...
+ration, measure and increase ration up to `1:1`. More threads than connections
+will be wasting server CPU.
 
-- missing new line characters at the end of messages
-- an invalid data format such as unescaped special characters
-- invalid column / table name characters
-- schema mismatch with existing tables
-- message size overflows on the input buffer
-- system errors such as no space left on the disk
+Another consideration is the number of tables updates concurrently.
```
Comment (javier) [inline docs/reference/api/ilp/tcp-receiver.md:24]: is it the number of tables updated?

**Review by javier — APPROVED** [review]

**Pull request merged**